normandy/recipes/migrations/0014_auto_20190228_1128.py

```diff
@@ -27,10 +30,26 @@ def add_extension_id(apps, schema_editor):
         arguments = json.loads(revision.arguments_json)
         url = arguments.get("addonUrl")
         filename = get_filename_from_url(url)
-        extension = Extension.objects.get(xpi=f"extensions/{filename}")
-        arguments["extensionApiId"] = extension.id
-        revision.arguments_json = json.dumps(arguments)
-        revision.save()
+        try:
+            extension = Extension.objects.get(xpi=f"extensions/{filename}")
+        except Extension.DoesNotExist:
+            failures.append(
+                {
+                    "filename": filename,
+                    "addon_url": arguments.get("addonUrl"),
+                    "revision_id": revision.id,
+                    "recipe_id": revision.recipe.id,
+                }
+            )
+        else:
+            arguments["extensionApiId"] = extension.id
+            revision.arguments_json = json.dumps(arguments)
+            revision.save()
+
+    if failures:
+        for failure in failures:
+            sys.stderr.write(f"{failure}")
```

Comment:
I think you need a newline here, since `sys.`stderr.write` isn't as nice as print.

```suggestion
            sys.stderr.write(f"{failure}\n")
```

Alternatively, I think you could do `print(failure, file=sys.stderr)`.